api/apps/api/src/modules/specification/application/__tests__/fixtures.ts

```diff
@@ -0,0 +1,76 @@
+import { v4 } from 'uuid';
+import { Test } from '@nestjs/testing';
+
+import { CqrsModule, EventBus, IEvent } from '@nestjs/cqrs';
+import { SpecificationRepository } from '../specification.repository';
+import { SubmitSpecificationHandler } from '../submit-specification.handler';
+
+import { SubmitSpecification } from '../../commands/submit-specification.command';
+import { SpecificationOperation } from '../../feature-config';
+import { SpecificationCandidateCreated } from '../../events/specification-candidate-created.event';
+import { Specification } from '../../specification';
+
+export const getFixtures = async () => {
+  const scenarioId = v4();
+  let specificationId: string;
+  const sandbox = await (
+    await Test.createTestingModule({
+      imports: [CqrsModule],
+      providers: [
+        {
+          provide: SpecificationRepository,
+          useClass: InMemorySpecificationRepo,
+        },
+        SubmitSpecificationHandler,
+      ],
+    }).compile()
+  ).init();
+
+  const events: IEvent[] = [];
+  const eventBus = sandbox.get(EventBus);
+  const handler = sandbox.get(SubmitSpecificationHandler);
+  const repo: InMemorySpecificationRepo = sandbox.get(SpecificationRepository);
+
+  eventBus.subscribe((event) => {
+    events.push(event);
+  });
+
+  return {
+    WhenSubmitsValidSpecification: async () =>
+      (specificationId = await handler.execute(
+        new SubmitSpecification({
+          scenarioId,
+          draft: true,
+          features: [
+            {
+              operation: SpecificationOperation.Stratification,
+              baseFeatureId: v4(),
+              againstFeatureId: v4(),
+            },
+          ],
+        }),
+      )),
+    ThenItSavesTheSpecification() {
+      expect(specificationId).toBeDefined();
+      expect(repo.getById(specificationId)).toBeDefined();
```

Comment:
I'd check the entire #memory

Reply:
@Dyostiq you mean if there isn't any other scenarioId or/and the specification shape as well?

Reply:
any other scenarioId I think

Reply:
https://github.com/Vizzuality/marxan-cloud/pull/403/commits/fd39d4aa4c60a847ba0489234863375575dd5b97